File: 0206-reverse-linked-list/0206-reverse-linked-list.c
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     struct ListNode *next;
 * };
 */
struct ListNode* reverseList(struct ListNode* head) {
    struct ListNode *slow = NULL;
    struct ListNode *current = head;
    struct ListNode *fast = NULL;
    if(head == NULL || head->next == NULL) {
        return head;
    }
    while(current != NULL) {
        fast = current->next;
        current->next = slow;
        slow = current;
        current = fast;
        
    }
    return slow;
}

File: 0206-reverse-linked-list/0206-reverse-linked-list.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode reverseList(ListNode head) {
        ListNode current = head;
        ListNode slow = null;
        ListNode fast = null;
        if(head == null || head.next == null) {
            return head;
        }
        while(current != null) {
            fast = current.next;
            current.next = slow;
            slow = current;
            current = fast;
        }
        head = slow;
        return head;
    }
}